### PR TITLE
rbac: add e2e test for rbac tcp policies.

### PR DESCRIPTION
**Pull request by yangminzhu**

### CODE REVIEW

File: tests/e2e/tests/pilot/testdata/rbac/v1alpha1/istio-rbac-rules.yaml.tmpl [hunk 1]
```diff
@@ -1,11 +1,12 @@
-# istio-rbac-rules.yaml to enforce access control to services
+# istio-rbac-rules.yaml to enforce access control for http services
```
Comment (liminw) [inline tests/e2e/tests/pilot/testdata/rbac/v1alpha1/istio-rbac-rules.yaml.tmpl]: for http and TCP services? I saw you have TCP test cases in this file.

File: tests/e2e/tests/pilot/istio_rbac_test.go [hunk 2]
```diff
@@ -56,55 +58,105 @@ func TestRBAC(t *testing.T) {
...
+				port = fmt.Sprintf(":%d", req.port)
+			}
+			expectStr := "deny"
+			if req.allow {
+				expectStr = "allow"
+			}
+			testName := fmt.Sprintf("%s from %s cluster->%s%s%s[%s]",
+				req.src, cluster, req.dst, req.path, port, expectStr)
+
+			runRetriableTest(t, cluster, testName, 30, func() error {
```
Comment (liminw) [inline tests/e2e/tests/pilot/istio_rbac_test.go]: 30 => defaultRetryBudget?

File: tests/e2e/tests/pilot/testdata/rbac/v1alpha1/istio-rbac-rules.yaml.tmpl (hunk 1 above)
Comment (yangminzhu) [reply tests/e2e/tests/pilot/testdata/rbac/v1alpha1/istio-rbac-rules.yaml.tmpl]: Done. I forgot to mention the tcp here.

File: tests/e2e/tests/pilot/istio_rbac_test.go (hunk 2 above)
Comment (yangminzhu) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: defaultRetryBudget is 10 but occasionally it takes about 25 seconds in my local environment to finish a single test case, after that each remaining test case finished in less 1 seconds. So I feel we need a larger retry budget.

Comment (liminw) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: Shall we increase the retry budget?

File: tests/e2e/tests/pilot/istio_rbac_test.go [hunk 3]
```diff
@@ -56,55 +58,105 @@ func TestRBAC(t *testing.T) {
...
+				reqPath := fmt.Sprintf("http://%s%s%s", req.dst, port, req.path)
+				if !req.allow && port != "" {
```
Comment (liminw) [inline tests/e2e/tests/pilot/istio_rbac_test.go]: port != "" means TCP service? Can the test case specify both "port" and "path"?

Another question is, do we need to check the success case for TCP service?

Comment (yangminzhu) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: Yes, `port != ""` means the request will go through the tcp rbac filter. We can specify both port and path, but that doesn't make too much sense as we cannot enforce any access control on the path when it's going through the tcp rbac filter. Actually when path is "", I guess it's default to "/".

We do check the success case for TCP service, note the first if statement `!req.allow && port != ""` is true when the it's a tcp request and we're expecting a deny result.

The next else statement in line 150 will cover both success/fail http requests and success tcp requests. In that case, we can check the response code directly as the test client is behaving like a HTTP server even for those tcp ports.

File: tests/e2e/tests/pilot/istio_rbac_test.go (hunk 2 above)
Comment (yangminzhu) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: Do you mean the `defaultRetryBudget`? Here I guess 30 should be enough.

File: tests/e2e/tests/pilot/istio_rbac_test.go (hunk 3 above)
Comment (liminw) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: Sounds good. Thanks for explanation.

File: tests/e2e/tests/pilot/istio_rbac_test.go (hunk 2 above)
Comment (liminw) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: Yes, I mean defaultRetryBudget.

Comment (yangminzhu) [reply tests/e2e/tests/pilot/istio_rbac_test.go]: we probably should not to change as existing tests works well with this value.

**Pull request merged**